hedera-node/hedera-mono-service/src/main/java/com/hedera/services/base/state/InMemoryStateImpl.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.services.base.state;
+
+import com.swirlds.common.merkle.MerkleNode;
+import com.swirlds.common.merkle.utility.Keyed;
+import com.swirlds.merkle.map.MerkleMap;
+import java.time.Instant;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * An implementation of {@link com.hedera.services.base.state.State} backed by a {@link MerkleMap},
+ * resulting in a state that is stored in memory.
+ *
+ * @param <K> The type of key for the state
+ * @param <V> The type of value for the state
+ */
+public class InMemoryStateImpl<K, V extends MerkleNode & Keyed<K>> extends StateBase<K, V> {
+    private final MerkleMap<K, V> merkle;
+    private final Instant lastModifiedTime;
+
+    InMemoryStateImpl(@NotNull final String stateKey, @NotNull final Instant lastModifiedTime) {
+        super(stateKey);
+        this.merkle = new MerkleMap<>();
+        this.lastModifiedTime = lastModifiedTime;
+    }
+
+    public InMemoryStateImpl(
+            @Nonnull String stateKey,
```

Comment:
We probably want to standardize on either `@Nonnull` _or_ `@NotNull`; and either always explicitly throw `NullPointerException` on a null argument; or always rely on `Objects.requireNonNull()`.

Reply:
Agree. fixed to use `javax.annotation.Nonnull` annotation and `Objects.requireNonNull`